crates/storage/provider/src/providers/snapshot/manager.rs

```diff
@@ -175,6 +175,7 @@ impl SnapshotProvider {
         };
 
         jar.delete()?;
+        self.update_index(segment, None)?;
```

Comment:
is this an unrelated bug?

Reply:
unrelated, but this change is needed for `stage drop` to work correctly

Reply:
so yes related lol